tools/topology/topology2/doc/CMakeLists.txt

```diff
@@ -0,0 +1,53 @@
+# SPDX-License-Identifier: BSD-3-Clause
+
+cmake_minimum_required(VERSION 3.13)
+
+project(SOF_TOPOLOGY2_DOC NONE)
+
+set(SOF_ROOT_SOURCE_DIRECTORY "${PROJECT_SOURCE_DIR}/..")
+
+set(top_srcdir "${SOF_ROOT_SOURCE_DIRECTORY}")
+set(top_bindir "${PROJECT_BINARY_DIR}")
+
+configure_file(
+	"${PROJECT_SOURCE_DIR}/sof.doxygen.in"
+	"${PROJECT_BINARY_DIR}/sof.doxygen"
+)
+
+file(GLOB_RECURSE topology2_sources "${SOF_ROOT_SOURCE_DIRECTORY}/*.conf")
+file(GLOB_RECURSE extra_sources "${PROJECT_SOURCE_DIR}/extra-contents/*.doxy")
+
+add_custom_command(
+	OUTPUT ${PROJECT_BINARY_DIR}/contents.doxy
+	COMMAND ${PROJECT_SOURCE_DIR}/topology2-generate-contents.sh ${PROJECT_BINARY_DIR}/contents.doxy
+	DEPENDS ${PROJECT_SOURCE_DIR}/topology2-generate-contents.sh
+	DEPENDS ${topology2_sources}
+	VERBATIM
+)
+
+add_custom_target("doc-contents"
+	DEPENDS ${PROJECT_BINARY_DIR}/contents.doxy
+)
+
+add_custom_command(
+	OUTPUT ${PROJECT_BINARY_DIR}/doxygen/html/index.html
+	COMMAND doxygen sof.doxygen
+	DEPENDS ${PROJECT_BINARY_DIR}/sof.doxygen
+	DEPENDS ${PROJECT_SOURCE_DIR}/topology2-filter.py
+	DEPENDS "doc-contents"
+	DEPENDS ${extra_sources}
+	DEPENDS ${topology2_sources}
```

Comment:
Does doxygen reads the *.conf files directly? I doubt it. If not directly then remove this line, having it indirectly in the other target is enough and clearer.

Reply:
Yes it does. The *.conf files are the topology2 files I am documenting with doxygen. Well doxygen does not read the files directly. It runs topology2-filter.py with the file name it wants to open as an argument, and topology2-filter.py reads the file and outputs the converted file - with enough C syntax for doxygen to have some understanging about it - to stdout (well that is what gather it must be doing, I have not read the doxygen sources, and this filter part is very thinly documented).

So yes, the resulting document very much depend on the *.conf files. The contents.doxy does too and the document depends on that too, so in that way the dependency is redundant, but if for some reason in the future we drop the contents page, the document is still depending on the *.conf sources.

Reply:
I forgot that `topology2-filter.py` is invoked from doxygen, thanks for the clarification!